mypy/build.py

```diff
@@ -1244,7 +1244,14 @@ def skipping_module(self, id: str, path: str) -> None:
 
     def add_ancestors(self) -> None:
         # All parent packages are new ancestors.
-        ancestors = []
+        ancestors = []  # type: List[str]
+        if self.path is not None:
+            _, name = os.path.split(self.path)
+            base, _ = os.path.splitext(name)
+            if '.' in base:
+                # This is just a weird filename, don't add anything
+                self.ancestors = ancestors
```

Comment:
I think it'd be clearer if this used `[]` as the value instead of `ancestors` (which at this point is just a complex way to spell `[]`). Then I'd like to reorder things so that this block comes *before* `ancestors` is initialized. It probably won't need the annotation then.